Sources/OpenAI/Private/Networking/AsyncHTTPClientAdapter.swift

```diff
@@ -0,0 +1,131 @@
+//
+//  AsyncHTTPClientAdapter.swift
+//  SwiftOpenAI
+//
+//  Created by Joe Fabisevich on 5/18/25.
+//
+
+import AsyncHTTPClient
+import Foundation
+import NIOCore
+import NIOFoundationCompat
+import NIOHTTP1
+
+/// Adapter that implements HTTPClient protocol using AsyncHTTPClient
+public class AsyncHTTPClientAdapter: HTTPClient {
```

Comment:
Maybe to reduce the scope of the change in this PR, what do you think of:
- scope this code to only Linux.
- replaces usages of `AsyncHTTPClientAdapter.createDefault()` by `HTTPClient.createDefault()`
- add
```swift
extension HTTPClient {
  static func createDefault() -> HttpClient {
    #if os(Linux)
    AsyncHTTPClientAdapter.createDefault()
    #else
    URLSessionHTTPClientAdapter()
    #endif
  }
```